interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,728 @@
+        {coeff_1_5:<75}    2
+        {coeff_6_10:<75}    3
+        {coeff_11_14:<60}                   4"""
+    )
+    five_coeff_line = "{0: #015.8E}{1: #015.8E}{2: #015.8E}{3: #015.8E}{4: #015.8E}"
+    four_coeff_line = "{0: #015.8E}{1: #015.8E}{2: #015.8E}{3: #015.8E}"
+    low_and_high_temp = "{0:<#10.3F}{1:<#10.3F}"
+    thermo_data: dict[str, str] = {}
+    min_temp = 1.0e30
+    max_temp = 0.0
+    for spec in species:
+        if len(spec.composition) > 4:
+            raise NotImplementedError(
+                f"More than 4 elements in a species is unsupported: {spec.name}"
+            )
+        atoms = ""
+        for atom, amount in spec.composition.items():
+            atoms += f"{atom:<2}{int(amount):>3d}"
+        fmt_data = {"name": spec.name}
+        fmt_data["atoms"] = atoms
+
+        input_data = spec.input_data
+        if input_data["thermo"]["model"] != "NASA7":
+            raise NotImplementedError(
+                f"Species '{spec.name}' has unsupported thermo model '"
+                f"{input_data['thermo']['model']}. Supported thermo models are: NASA7"
+            )
+
+        if spec.thermo.min_temp < min_temp:
+            min_temp = spec.thermo.min_temp
+        if spec.thermo.max_temp > max_temp:
+            max_temp = spec.thermo.max_temp
+
+        note = input_data["thermo"].get("note", "")
+        if len(note) <= 6:
+            comment = ""
+            fmt_data["note"] = note
+        else:
+            comment = note
+            fmt_data["note"] = ""
+        temperature_range = input_data["thermo"]["temperature-ranges"]
+        if len(temperature_range) == 3:
+            fmt_data["low_and_high_temp"] = low_and_high_temp.format(
+                temperature_range[0], temperature_range[2]
+            )
+            fmt_data["mid_temp"] = f"{temperature_range[1]:<#8.3F}"
+        elif len(temperature_range) == 2:
+            fmt_data["low_and_high_temp"] = low_and_high_temp.format(*temperature_range)
+            fmt_data["mid_temp"] = f"{' ':8}"
+        else:
+            raise NotImplementedError(
+                f"Species '{spec.name}' has more than 2 temperature ranges. Only "
+                "1 or 2 are supported."
+            )
+        low_coeffs = input_data["thermo"]["data"][0]
+        high_coeffs = input_data["thermo"]["data"][1]
+        fmt_data["coeff_1_5"] = five_coeff_line.format(*high_coeffs[:5])
+        fmt_data["coeff_6_10"] = five_coeff_line.format(
+            *(high_coeffs[5:] + low_coeffs[:3])
+        )
+        fmt_data["coeff_11_14"] = four_coeff_line.format(*low_coeffs[3:])
+        text = fmt.format(**fmt_data)
+        if comment:
+            text = f"!{comment}\n" + text
+        thermo_data[spec.name] = text
+
+    if separate_file:
+        leader = "THERMO"
+    else:
+        leader = "THERMO ALL"
+
+    return (
+        f"{leader}\n{min_temp:.3F}   1000.000  {max_temp:.3F}\n\n"
```

Comment:
The thread is here: https://github.com/Cantera/cantera/pull/1185#discussion_r828649674 The problem is that CHEMKIN kinda sorta uses the McBride NASA format, but also has some changes that mean the values here are never actually used for anything, since each species is required to have its own definition of all three temperatures according to CHEMKIN (general NASA programs do not require that, apparently?). Anyhow, I'm not sure what to say here that would be coherent and helpful for a future reader... I'll try to add something useful 😄

Reply:
That would be great!

Reply:
Could you add this comment?

Reply:
Thanks for the reminder, done now.